src/components/landing/pages/login/auth/via-internal.tsx

```diff
@@ -0,0 +1,84 @@
+import React, { FormEvent, useState } from 'react'
+import { Alert, Button, Card, Form } from 'react-bootstrap'
+import { Trans, useTranslation } from 'react-i18next'
+import { useSelector } from 'react-redux'
+import { Link } from 'react-router-dom'
+import { doInternalLogin } from '../../../../../api/auth'
+import { ApplicationState } from '../../../../../redux'
+import { getAndSetUser } from '../../../../../utils/apiUtils'
+import { ShowIf } from '../../../../common/show-if/show-if'
+
+export const ViaInternal: React.FC = () => {
+  const { t } = useTranslation()
+  const [username, setUsername] = useState('')
+  const [password, setPassword] = useState('')
+  const [error, setError] = useState(false)
+  const allowRegister = useSelector((state: ApplicationState) => state.config.allowRegister)
+
+  const doAsyncLogin = async () => {
+    await doInternalLogin(username, password)
+    await getAndSetUser()
+  }
```

Comment:
You don't need this function.

Reply:
This function was already there when I started editing the code. Another similar function exists in the via-ldap component. I'll remove/refactor it there too.

Reply:
Done